### PR TITLE
Question.choices could use separator from setttings.CHOICES_SEPARATOR

### PR DESCRIPTION
**Pull request by Fibio**
Hi I met the problem when my question.choices contain comma, so split choices doesn't work correctly, I hope my update will be helpful for other developers, thanks for your job.

Best, 
Tanya

### CODE REVIEW

File: survey/forms.py [hunk 1]
```diff
@@ -99,7 +100,9 @@ def get_question_initial(self, question, data):
                 elif "[" in answer.body and "]" in answer.body:
                     initial = []
                     unformated_choices = answer.body[1:-1].strip()
-                    for unformated_choice in unformated_choices.split(","):
+                    for unformated_choice in unformated_choices.split(
+                        getattr(settings, "CHOICES_SEPARATOR", ",")
```
Comment (Pierre-Sassoulas) [inline survey/forms.py]: Would it work by using `settings.CHOICES_SEPARATOR` here and deporting the getattr in `survey/settings.py` ? I don't remember that well but I think every optional settings was done that way.

File: survey/tests/models/test_question.py [hunk 2]
```diff
@@ -71,6 +72,32 @@ def test_get_choices(self):
             self.questions[0].get_choices(), (("聖黎", "聖黎"), ("はじむ", "はじむ"))
         )
 
+    @override_settings(CHOICES_SEPARATOR="|")
```
Comment (Pierre-Sassoulas) [inline survey/tests/models/test_question.py]: Very nice !

File: survey/models/question.py [hunk 3]
```diff
@@ -106,7 +106,7 @@ def get_clean_choices(self):
         if self.choices is None:
             return []
         choices_list = []
-        for choice in self.choices.split(","):
```
Comment (Pierre-Sassoulas) [inline survey/models/question.py]: This was actually a magic string, nice catch! Also, it permits to use comma in question by using an exotic separator and that was not possible before. Very nice change !

File: survey/forms.py (hunk 1 above)
Comment (Fibio) [reply survey/forms.py]: yes, I just added it with case when CHOICES_SEPARATOR could be not set in setting at all

File: survey/models/question.py (hunk 3 above)
Comment (Fibio) [reply survey/models/question.py]: So do you apply this? I'm not sure about my tests naming according to Codacy/PR Quality Review, where am I wrong here?

Comment (Fibio) [reply survey/models/question.py]: oh just saw you main comment, thanks

**Comment by Fibio** [comment]
So shell I change tests naming and remove getattr?

**Comment by Pierre-Sassoulas** [comment]
Getattr need to be used only once in the survey/settings.py file so it has
only one default value. But if the function name with 'with_pipe' is still
too long for codacy I'll merge it with your current name this warning is
not really important and we should not waste too much time on it :)

Thank you again !

File: survey/forms.py (hunk 1 above)
Comment (Fibio) [reply survey/forms.py]: done

**Comment by Fibio** [comment]
Thanks for you review, it was really helpful

**Comment by Fibio** [comment]
Done

**Comment by Pierre-Sassoulas** [comment]
Thank you very much for the PR, I'm going to release a new version with your changes.

**Pull request merged**